SeaPublicWebsite/Controllers/EnergyEfficiencyController.cs

```diff
@@ -1316,12 +1316,24 @@ await googleAnalyticsService.SendEvent(new GaRequestBody
         propertyData.ReturningUser = true;
         await propertyDataStore.SavePropertyDataAsync(propertyData);
 
-        var recommendations = propertyData.PropertyRecommendations;
+        // Always request the latest recommendations
+        var updatedPropertyData =
+            await propertyDataService.UpdatePropertyDataWithRecommendations(reference);
+
+        var recommendations = updatedPropertyData.PropertyRecommendations;
+
+        // On any change we show the user this page which explains why their recommendations may have changed
+        if (propertyData.RecommendationsUpdatedSinceLastVisit)
```

Comment:
is this propertyData up to date with the changes from the call to UpdatePropertyDataWithRecommendations?

It looks like we fetch `propertyData` in line 1312, but then save the update in recs to `updatedPropertyData`. Is this looking at the old version of `propertyData` intentionally?

Reply:
ah good spot and this should reference `updatedPropertyData`. it does work as this (I suppose EF magic is updating `propertyData`?) though more clear to use the updated variable

Reply:
Thanks, thought as much re. EF - probably some magic behind the scenes model caching makes it stay up to date somehow.